Precommit lint doesn't ignore create-app template

There is currently an ignore rule for the template inside create-app package:
https://github.com/spotify/backstage/blob/2cd0904e3594fb73cef5dacdd382a9af26923615/packages/create-app/.eslintrc.js#L3

But it doesn't get picked up by precommit lint script, which causes errors due to `package.json` being a template (package.json.hbs)

## Expected Behavior
No lint checks inside create-app's template folder (or at least some rules disabled)


## Current Behavior
Errors due to package.json formally missing inside template folder

## Possible Solution

<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas as to the implementation of the addition or change -->

## Steps to Reproduce
1. Change code inside packages/create-app/templates/default-app/packages/app/src
2. Try to commit it


## Context
```
husky > pre-commit (node v12.16.2)
tasks...
  ❯ Running tasks for *.{js,jsx,ts,tsx}
    ✖ eslint --fix [FAILED]
    ◼ prettier --write
  ↓ No staged files match *.{json,md} [SKIPPED]
↓ Skipped because of errors from tasks. [SKIPPED]
✔ Reverting to original state because of errors...
✔ Cleaning up... 

✖ eslint --fix:

backstage/packages/create-app/templates/default-app/packages/app/src/App.tsx
   1:27  error  'react' should be listed in the project's dependencies. Run 'npm i -S react' to add it                                            import/no-extraneous-dependencies
   7:8   error  '@backstage/core' should be listed in the project's dependencies. Run 'npm i -S @backstage/core' to add it                        import/no-extraneous-dependencies
  11:41  error  'react-router' should be listed in the project's dependencies. Run 'npm i -S react-router' to add it                              import/no-extraneous-dependencies
  12:41  error  '@backstage/plugin-catalog' should be listed in the project's dependencies. Run 'npm i -S @backstage/plugin-catalog' to add it    import/no-extraneous-dependencies
  13:38  error  '@backstage/plugin-techdocs' should be listed in the project's dependencies. Run 'npm i -S @backstage/plugin-techdocs' to add it  import/no-extraneous-dependencies

✖ 5 problems (5 errors, 0 warnings)

husky > pre-commit hook failed (add --no-verify to bypass)
```

--- a/.eslintignore
+++ b/.eslintignore
@@ -7,3 +7,5 @@
 **/.git/**
 **/public/**
 **/microsite/**
+**/templates/**
+**/sample-templates/**

